ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -336,6 +336,63 @@ function upgrade() {
         clear_upgrade
         check_upgrade_status "${cluster_src_ver}"
     fi
+    if check_ota_case_enabled "OCP-25473"; then
+        local retry=0 previous_image output PREVIOUS_MINOR_VERSION
+        previous_image="$(< "${SHARED_DIR}/upgrade-edge")"
+        if [[ -n "${previous_image:-}" ]]; then
+            run_command "oc adm upgrade --to-image=${previous_image} --allow-explicit-upgrade --force=${FORCE_UPDATE}"
+        else
+            echo "Error: The previous image is not given, break the job"
+            exit 1
+        fi
+        
+        echo "OCP-25473: Waiting for the first time upgrade get to start"
+        while [[ "$(oc get clusterversion version -o json|jq '.status.conditions[]|select(.type=="Progressing")'|jq -r '.status')" != 'True' ]]; do
+            if [[ retry -ge 5 ]]; then
+                echo "Error: OCP-25473: First time upgrade not started, break the job"
+                exit 1
+            fi
+            retry=$(( retry+1 ))
+            sleep 1m
+        done
+        echo "OCP-25473: Upgrade is processing"
+
+        echo "OCP-25473: Upgrade again without --allow-upgrade-with-warnings"
+        oc adm upgrade --to-image=${TARGET} --allow-explicit-upgrade --force=${FORCE_UPDATE} > ${ARTIFACT_DIR}/ota_25473_result 2>&1 || true
+        output="$(cat ${ARTIFACT_DIR}/ota_25473_result)"
+        if [[ ! "${output}" =~ "error: the cluster is already upgrading" ]] || [[ ! "${output}" =~ "If you want to upgrade anyway, use --allow-upgrade-with-warnings" ]]; then
+            echo "Error: OCP-25473:  Re-upgrade should not start and raise error when there is already an upgrade is processing"
+            echo "The output of 'oc adm upgrade':"
+            echo $output
+            exit 1
+        fi
+        
+        echo "OCP-25473: Upgrade again with --allow-upgrade-with-warnings"
+        # oc adm upgrade --to-image=${TARGET} --allow-explicit-upgrade --allow-upgrade-with-warnings=true --force=${FORCE_UPDATE} > ${ARTIFACT_DIR}/ota_25473_result 2>&1 || true
+        # output="$(cat ${ARTIFACT_DIR}/ota_25473_result)"
+        PREVIOUS_MINOR_VERSION=$(oc get clusterversion version -ojson | jq -r '.status.desired.version' | cut -f2 -d.)
+        run_command "oc adm upgrade --to-image=${TARGET} --allow-explicit-upgrade --allow-upgrade-with-warnings=true --force=${FORCE_UPDATE}"
+        if [[ "$PREVIOUS_MINOR_VERSION" == "$TARGET_MINOR_VERSION" ]]; then
+            if [[ "$SOURCE_MINOR_VERSION" == "$TARGET_MINOR_VERSION" ]]; then
+                # e.g.: minor verion 1 = minor verion 2 = minor verion 3
+                echo "Re-run Z Stream upgrade"
+                echo "Upgrading cluster to ${TARGET} gets started..."
+            else
+                # e.g.: minor verion 1 < minor verion 2 = minor verion 3
+                echo "Re-run Y Stream upgrade, how do we deal with this kind of upgrade? OTA-861"
+            fi
+        else
```

Comment:
@evakhoni But in this conditional branch， we should not upgrade to next Y version except when we using `--force`, right?

Reply:
as mentioned over slack, a retarget to y-stream using force is not supported, and will definitely break the cluster in some cases, as such it should not be tested.

Reply:
I should also clarify, since a retarget using force should not be done, the Target version should always be Signed. running this test while retargetting to a nightly should not be allowed.

Reply:
@evakhoni If I understand correctly, you mean re-target to a new Y stream is not allowed, but re-target to a new z stream is allowed. right? Here no matter a nightly or stable build.
As I tested in below jobs, re-target to z stream is allowed:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58136/rehearse-58136-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-baremetal-ipi-ovn-ipv4-arm-f28/1852196454089625600

and 

https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/58136/rehearse-58136-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-baremetal-ipi-ovn-ipv4-fips-vmedia-amd-f28/1852196481340018688

Reply:
issue is, for nightly we're forced to use --force, so we'll be unable to test if z-stream retarget is actually accepted by the cluster, or if it is broken but forced, because with force it will update even if it not supposed to.